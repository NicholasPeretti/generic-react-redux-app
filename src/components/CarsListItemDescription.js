import React from 'react'
import PropTypes from 'prop-types'
import CarTitle from './CarTitle'
import CarShortDescription from './CarShortDescription'
import StyleContext from '../styleContext'

var CarsListItemDescription = ({ car }) => (
  <StyleContext.Consumer>
    {context => {
      const style = {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-between',
        fontSize: '14px',
        margin: context.spacing.padding2
      }
      return (
        <section style={style}>
          <CarTitle
            manufacturerName={car.manufacturerName}
            modelName={car.modelName}
          />
          <CarShortDescription {...car} />
          <a href='#'>View details</a>
        </section>
      )
    }}
  </StyleContext.Consumer>
)

CarsListItemDescription.propTypes = {
  car: PropTypes.shape({
    stockNumber: PropTypes.number.isRequired,
    manufacturerName: PropTypes.string.isRequired,
    modelName: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    mileage: PropTypes.shape({
      number: PropTypes.number.isRequired,
      unit: PropTypes.string.isRequired
    }).isRequired,
    fuelType: PropTypes.string.isRequired,
    pictureUrl: PropTypes.string.isRequired
  }).isRequired
}

export default CarsListItemDescription
