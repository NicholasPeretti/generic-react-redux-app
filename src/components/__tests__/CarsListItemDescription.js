import React from 'react'
import { shallow } from 'enzyme'
import CarsListItemDescription from '../CarsListItemDescription'
import PresentationalTest from '../../utils/tests/presentational'

const props = {
  car: {
    stockNumber: 86987,
    manufacturerName: 'Fiat',
    modelName: 'Talento',
    color: 'white',
    mileage: {
      number: 102586,
      unit: 'km'
    },
    fuelType: 'Petrol',
    pictureUrl: 'http://localhost:3001/car.svg'
  }
}

describe('CarsListItemDescription', () => {
  PresentationalTest(CarsListItemDescription, props)

  const wrapper = shallow(<CarsListItemDescription {...props} />)

  test('Should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
