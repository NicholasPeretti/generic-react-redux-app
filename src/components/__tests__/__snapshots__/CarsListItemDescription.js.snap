// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarsListItemDescription Should match snapshot 1`] = `"<section style=\\"display:flex;flex-direction:column;justify-content:space-between;font-size:14px;margin:12px\\"><h3 style=\\"margin:0px\\">Fiat Talento</h3><p style=\\"margin:0px\\">Stock # 86987 - 102586 km - Petrol - white</p><a href=\\"#\\">View details</a></section>"`;
